Content.Server/Botany/Systems/PlantAnalyzerSystem.cs

```diff
@@ -0,0 +1,287 @@
+        tolerances = CheckAllTolerances(comp, tolerances);
+
+        generalTraits = CheckGeneralTraits(comp, generalTraits);
+
+        return new PlantAnalyzerScannedSeedPlantInformation(GetNetEntity(target), seedName,
+            seedChem, plantHarvestType, exudeGases, plantMutations, false, plantSpeciation, tolerances.ToString(), generalTraits.ToString(), _scanMode);
+    }
+
+    /// <summary>potency.ToString(), yield.ToString(),
+    ///     Analysis of unique seed.
+    /// </summary>
+    public PlantAnalyzerScannedSeedPlantInformation ObtainingGeneDataSeed(SeedData comp, EntityUid target, bool trayChecker)
+    {
+        var seedName = Loc.GetString(comp.DisplayName);
+
+        var seedChem = "\r\n   ";
+        var plantHarvestType = "";
+        var exudeGases = "";
+
+        var plantSpeciation = "";
+        var plantMutations = "";
+        var tolerances = "";
+        var generalTraits = "";
+
+        if (comp.HarvestRepeat == HarvestType.Repeat) plantHarvestType = Loc.GetString("plant-analyzer-harvest-repeat");
+        if (comp.HarvestRepeat == HarvestType.NoRepeat) plantHarvestType = Loc.GetString("plant-analyzer-harvest-ephemeral");
+        if (comp.HarvestRepeat == HarvestType.SelfHarvest) plantHarvestType = Loc.GetString("plant-analyzer-harvest-autoharvest");
+
+        seedChem += string.Join("\r\n   ", comp.Chemicals.Select(item => item.Key.ToString()));
+
+        if (comp.ExudeGasses.Count > 0)
+        {
+            foreach (var (gas, amount) in comp.ExudeGasses)
+            {
+                exudeGases += "\r\n   " + gas;
+            };
+        }
+        else
+        {
+            exudeGases = Loc.GetString("plant-analyzer-plant-gasses-no");
+        }
+
+        if (_scanMode)
+        {
+            plantSpeciation += String.Join(", \r\n", comp.MutationPrototypes.Select(item => item.ToString()));
+            plantMutations = CheckAllMutations(comp, plantMutations);
+            tolerances = CheckAllTolerances(comp, tolerances);
+        }
+
+        generalTraits = CheckGeneralTraits(comp, generalTraits);
+
+        return new PlantAnalyzerScannedSeedPlantInformation(GetNetEntity(target), seedName,
+            seedChem, plantHarvestType, exudeGases, plantMutations, false, plantSpeciation, tolerances.ToString(), generalTraits.ToString(), _scanMode);
+    }
+
+    /// <summary>
+    ///     Returns information about the seed.
+    /// </summary>
+    public string CheckGeneralTraits(SeedData plant, string generalTraits)
+    {
+        var name = Loc.GetString(plant.DisplayName);
+        var endurance = plant.Endurance;
+        var yield = plant.Yield;
+        var lifespan = plant.Lifespan;
+        var maturation = plant.Maturation;
+        var growthStages = plant.GrowthStages;
+        var potency = plant.Potency;
+
+        generalTraits = name + ";" + endurance + ";" + yield.ToString() + ";" + lifespan + ";" + maturation + ";" + growthStages + ";" + potency;
```

Comment:
Instead of sending strings and requiring parsing on the other end, it would be better to send a `List<String>`

Reply:
Only to replace the arrays I used or use a single List<String> as Parameter for PlantAnalyzerScannedSeedPlantInformation ?

Reply:
I went back to sending as individual parameters, and only the boolean mutations are in a list.